crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -298,30 +300,36 @@ impl ShaderCache {
     }
 }
 
+type LayoutCacheKey = (Vec<BindGroupLayoutId>, Vec<PushConstantRange>);
 #[derive(Default)]
 struct LayoutCache {
-    layouts: HashMap<Vec<BindGroupLayoutId>, ErasedPipelineLayout>,
+    layouts: PreHashMap<LayoutCacheKey, ErasedPipelineLayout>,
```

Comment:
I don't think this improves anything. See https://github.com/bevyengine/bevy/pull/4908#discussion_r1107723852